ccp/modules/datashield-compose.yml

```diff
@@ -0,0 +1,103 @@
+services:
+  ############################################ DataSHIELD Client (Rocker R-Studio)
+  rstudio:
+    container_name: bridgehead-rstudio
+    image: docker.verbis.dkfz.de/ccp/dktk-rstudio:latest
+    environment:
+      USER: "ruser"
+      PASSWORD: "${RSTUDIO_PASSWORD}"
+      HTTP_RELATIVE_PATH: "/rstudio"
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.rstudio_ccp.rule=PathPrefix(`/rstudio`)"
+      - "traefik.http.services.rstudio_ccp.loadbalancer.server.port=8787"
+      - "traefik.http.routers.rstudio_ccp.tls=true"
+      - "traefik.http.middlewares.rstudio_ccp_strip.stripprefix.prefixes=/rstudio"
+      - "traefik.http.routers.rstudio_ccp.middlewares=rstudio_ccp_strip"
+      #volumes:
+      #- "bridgehead-rstudio-config:/home/rstudio/.config/rstudio"
+      #- "bridgehead-rstudio-workspace:/home/rstudio/workspace"
+
+  ############################################ DataSHIELD Server (Opal)
+  opal:
+    container_name: bridgehead-opal
+    image: docker.verbis.dkfz.de/ccp/dktk-opal:latest
+    labels:
+      - "traefik.enable=true"
+      - "traefik.http.routers.opal_ccp.rule=PathPrefix(`/opal`)"
+      - "traefik.http.services.opal_ccp.loadbalancer.server.port=8080"
+      - "traefik.http.routers.opal_ccp.tls=true"
+    links:
+      - opal-rserver
+      - opal-db
+    environment:
+      JAVA_OPTS: "-Xms1G -Xmx8G -XX:+UseG1GC"
+      # OPAL_ADMINISTRATOR_USER: "administrator" # Please, don't remove this line. It is informative.
+      OPAL_ADMINISTRATOR_PASSWORD: "${OPAL_ADMINISTRATOR_PASSWORD}"
+      POSTGRESDATA_HOST: "opal-db"
+      POSTGRESDATA_DATABASE: "opal"
+      POSTGRESDATA_USER: "opal"
+      POSTGRESDATA_PASSWORD: "${OPAL_DB_PASSWORD}"
+      ROCK_HOSTS: "opal-rserver:8085"
+      APP_URL: "https://${HOST}/opal"
+      APP_CONTEXT_PATH: "/opal"
+      OPAL_PRIVATE_KEY: "/run/secrets/opal-key.pem"
+      OPAL_CERTIFICATE: "/run/secrets/opal-cert.pem"
```

Comment:
Are we sure this is needed? Won't this be accessed via plain http, and only in this stack, anyway?

Reply:
See comments of @Threated

Reply:
Opal sadly does not provide **any** way to talk over plain http so we need to a tls connection and therefore these certificates.